lib/core/paypal_http_client.rb

```diff
@@ -25,15 +25,17 @@ def user_agent
       "PayPalSDK/rest-sdk-ruby #{VERSION} (#{library_details})"
     end
 
+    private
+
     def _sign_request(request)
-      if (!_has_auth_header(request) && !_is_auth_request(request))
-        if (!@access_token || @access_token.isExpired)
-          accessTokenRequest = PayPal::AccessTokenRequest.new(@environment, @refresh_token)
-          tokenResponse = execute(accessTokenRequest)
-          @access_token = PayPal::AccessToken.new(tokenResponse.result)
-        end
-        request.headers["Authorization"] = @access_token.authorizationString()
+      return if _has_auth_header(request) || _is_auth_request(request)
```

Comment:
i'm pretty sure this is supposed to be `&&` and not `||`

Reply:
I believe this is the correct way of writing it and sounds like something coming from RuboCop’s autocorrect.
Since this is an early return, it seems to me that we want the inverse of `if (!a && !b)`, which is `if (a || b)`.

Another way of writing it would be using `unless`:

```suggestion
      return unless !_has_auth_header(request) && !_is_auth_request(request)
```